### PR TITLE
change card tree in 7.x to respect sort order

### LINKED ISSUE #8187: Card Tree is not respecting sort order in 7.x
**Describe the bug**

The card tree in the editor is not respecting sort order in 7.x.

**To Reproduce**
1. Go to any created resource
2. The card order does not match what it setup in the graph designer.

**Screenshots**

**Expected behavior**

**Your Arches Information**

*   Version used:
*   Operating System and version (desktop or mobile):
*   Browser Name and version:
*   Link to your Arches Install (optional):

**Additional context**

**Ticket Background**
*   Found by: @

### PR DESCRIPTION
**Pull request by aarongundel**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

The card tree in 7.x no longer appears to respect sort order.  This may be due to graph publication changes.  This fixes that at the client level.

### Issues Solved

fixes #8187 

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: 
*   Found by: @ 
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

**Comment by aarongundel** [comment]
This issue is dependent on fix for #8172, that one should merge first.

File: arches/app/media/js/views/graph/graph-designer/card-tree.js [hunk 1]
```diff
@@ -391,6 +391,8 @@ define([
                 return ko.unwrap(group.nodegroupid) === card.nodegroup_id;
             });
             return !nodegroup || !ko.unwrap(nodegroup.parentnodegroup_id);
+        }).sort((firstEl, secondEl) => {
+            return firstEl.sortorder >= secondEl.sortorder ? firstEl.sortorder === secondEl.sortorder ? 0 : 1 : -1;
```
Comment (chrabyrd) [inline arches/app/media/js/views/graph/graph-designer/card-tree.js:395]: can the compound ternary be rewritten?

File: arches/app/media/js/views/resource/editor.js [hunk 2]
```diff
@@ -85,6 +85,8 @@ define([
             return group.nodegroupid === card.nodegroup_id;
         });
         return !nodegroup || !nodegroup.parentnodegroup_id;
+    }).sort((firstEl, secondEl) => {
+        return firstEl.sortorder >= secondEl.sortorder ? firstEl.sortorder === secondEl.sortorder ? 0 : 1 : -1;
```
Comment (chrabyrd) [inline arches/app/media/js/views/resource/editor.js:89]: can the compound ternary be rewritten?

**Review by chrabyrd — COMMENTED** [review]
cool just gave this a first pass, will check more in-depth once the blocking branch has been merged in.

File: arches/app/models/graph.py [hunk 3]
```diff
@@ -1377,14 +1377,17 @@ def find_unpermitted_edits(obj_a, obj_b, ignore_list, obj_type):
...
+                    try:
+                        unpermitted_node_edits = find_unpermitted_edits(
+                            db_node,
+                            self.nodes[db_node.nodeid],
+                            ["name", "issearchable", "ontologyclass", "description", "isrequired", "fieldname", "exportable"],
+                            "node",
+                        )
+                        if unpermitted_node_edits is not None:
+                            unpermitted_edits.append(unpermitted_node_edits)
+                    except KeyError:
```
Comment (chrabyrd) [inline arches/app/models/graph.py:1389]: why `pass` on KeyError here?

File: arches/app/media/js/views/graph/graph-designer/card-tree.js [hunk 4]
```diff
@@ -391,6 +391,16 @@ define([
...
+        }).sort((firstEl, secondEl) => {
+            if(firstEl.sortorder >= secondEl.sortorder) {
+                if(firstEl.sortorder === secondEl.sortorder) {
+                    return 0;
+                } else {
+                    return 1;
+                }
+            } else {
+                return -1;
+            }
```
Comment (chrabyrd) [inline arches/app/media/js/views/graph/graph-designer/card-tree.js:403]: nit: this doesn't need to be a nested if-statement

File: arches/app/media/js/views/resource/editor.js [hunk 5]
```diff
@@ -85,6 +85,16 @@ define([
...
+    }).sort((firstEl, secondEl) => {
+        if(firstEl.sortorder >= secondEl.sortorder) {
+            if(firstEl.sortorder === secondEl.sortorder) {
+                return 0;
+            } else {
+                return 1;
+            }
+        } else {
+            return -1;
+        }
```
Comment (chrabyrd) [inline arches/app/media/js/views/resource/editor.js:97]: nit: this doesn't need to be a nested if-statement

File: arches/app/models/graph.py [hunk 6]
```diff
@@ -1586,15 +1589,18 @@ def publish(self, notes=None):
...
+                serialized_graph=JSONDeserializer().deserialize(JSONSerializer().serialize(self, force_recalculation=True)),
+                notes=notes,
+            )
+            publication.save()
 
-        self.publication = publication
-        self.save()
+            self.publication = publication
+            self.save()
+        except Exception:
```
Comment (chrabyrd) [inline arches/app/models/graph.py:1602]: under what circumstance would this exception be hit?

**Comment by aarongundel** [comment]
@chrabyrd I've fixed the nesting issue you originally noted; I agree - I hate excessive nesting.  The other issues re: try/catch went away on merging 7.x in, as there were lots of deletes in the 7.x branch.  I suspect they were originally there because I was running into some obtuse issue that is no longer relevant.  

These now sort the trees as desired - but the saving of the sort is another matter that may be worthy of another ticket (re ordering the card tree does not provide an option for save).  Low priority issue though.

**Comment by chrabyrd** [comment]
wow you weren't kidding about the thing getting cleaned up on merge. lgtm! 👍

**Pull request merged**